repos/system_upgrade/el7toel8/actors/cupsmigrate/actor.py

```diff
@@ -0,0 +1,339 @@
+          cupsd_conf.write(cupsd_contents)
+        return True
+      except IOError:
+        return False
+
+    def move(self, directives, source, target):
+      """
+      move()
+  
+      parameters:
+      - directives - list of strings
+      - source - string
+      - target - string
+  
+      returns boolean
+      """
+
+      try:
+        with open(source, 'r') as opened_source:
+          source_content = opened_source.read()
+      except IOError:
+        return False
+
+      try:
+        with open(target, 'r') as opened_target:
+          target_content = opened_target.read()
+      except IOError:
+        return False
+
+      # grap directive and its value from old config and put it into new config
+      for directive in directives:
+        regexp = '(' + directive + '\s+.+\s*\n?)'
+        pattern = compile(regexp)
+        found_lines = findall(pattern, source_content)
+        for line in found_lines:
+          try:
+            with open(target, 'w') as opened_target:
+              opened_target.write(target_content + line)
+          except IOError:
+            return False
+
+          target_content = target_content + line
+          source_content = source_content.replace(line, '')
+          try:
+            with open(source, 'w') as opened_source:
+              opened_source.write(source_content)
+          except IOError:
+            return False
+
+      return True
+
+    def process(self):
+      for model in self.consume(Cupsmigrationmodel):
+        if model.migrateable is False:
+          report_generic(
+                          title='CUPS will not be migrated.',
+                          summary='CUPS is not installed or its configuration is up-to-date.',
+                          severity='low'
+                        )
+          raise StopActorExecution('CUPS will not be migrated')
+
+        include_files = model.include_files
+        interface = model.migration_matrix.interface
+        digest = model.migration_matrix.digest
+        include = model.migration_matrix.include
+        certkey = model.migration_matrix.certkey
+        env = model.migration_matrix.env
+        printcap = model.migration_matrix.printcap
+
+
+        if interface is True:
+          report_generic(
+                          title='Print queues with interface scripts are used on the machine.',
+                          summary='There are files in /etc/cups/interfaces directory, which indicates usage of print queues with interface scripts. Interface script were removed from CUPS due security issues, so print queues with them will not work in newer CUPS version. Please try to install your printer with printer driver.',
+                          severity='high'
+                        )
+        if include is True:
+          result = self.migrate_includes(include_files)
+          if result is not True:
+            raise StopActorExecutionError('Include directive could not be migrated due IOError')
```

Comment:
My other comments about functions returning meaningful error messages is related with this raised exception (and following ones). This error will stop Leapp processing (not only this actor), if that is what we want we should make it clear why it failed and how to solve the problem.

Reply:
Removed.